--- xsysinfo.man.org	2005-12-03 20:41:59.000000000 +0100
+++ xsysinfo.man	2005-12-03 20:45:13.000000000 +0100
@@ -1,4 +1,4 @@
-.\" Copyright (c) 1999  Roland Rosenfeld <roland@debian.org>
+.\" Copyright (c) 1999-2005  Roland Rosenfeld <roland@debian.org>
 .\"               
 .\" This manual page is free software; you can redistribute it and/or modify
 .\" it under the terms of the GNU General Public License as published by
@@ -12,18 +12,18 @@
 .\" 
 .\" You should have received a copy of the GNU General Public License
 .\" along with this program; if not, write to the Free Software
-.\" Foundation, Inc.,59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+.\" Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
 .\"
 .\" This manual page is written especially for Debian Linux.
 .\"
-.TH XSYSINFO 1x "June 1999" "Debian Project" "Debian GNU/Linux"
+.TH XSYSINFO 1x "December 2005" "Debian Project" "Debian GNU/Linux"
 .SH NAME
 xsysinfo \- Display Linux kernel parameters in graphical form
 .SH SYNOPSIS
 .B xsysinfo [-help] [-update
 .IB n ]
 .B [-[no]title] [-[no]labels] [-[no]loadavg] [-[no]load] [-[no]mem]
-.B [-[no]swap] 
+.B [-[no]swap] [-[no]smp]
 .SH DESCRIPTION
 .B Xsysinfo
 is an X application to display some Linux kernel parameters in
@@ -36,7 +36,7 @@
 .BI -update " n"
 Set update rate to
 .I n
-seconds
+milli-seconds
 .TP
 .B -title
 Show title string
